### ISSUE TITLE
Please fix postfix::relayhost parameter

### ISSUE THREAD

**Issue by XMol** [camptocamp/puppet-postfix#295]
Hi,

as was already reported several years ago (#187), the `postfix::relayhost` parameter is meaningless. Unlike many other parameters, it is not looked up by `postfix::files` and therefor has no effect, which we discovered today.

I'd suggest to either remove the parameter and force people to include it in `postfix::configs` or to define `Postfix::Config['relayhost']`  themselves, or handle it in `postfix::files`, same as with [four more parameters](https://github.com/camptocamp/puppet-postfix/blob/master/manifests/files.pp#L101).

Thank you,  
Xavier.

**Comment by raphink** [comment]
It seems that `postfix::relayhost` is used in `postfix::mta`:

https://github.com/camptocamp/puppet-postfix/blob/master/manifests/mta.pp#L40-L45

**Comment by XMol** [comment]
True, but we don't have `postfix::mta` in our catalog, which seemingly isn't mandatory. Including that - or `postfix::satellite`, which does the same - just to have `relayhost` included in `main.cf` cannot be the advised solution, right?

**Comment by DLeich** [comment]
I just got bit by this too.  Was scratching my head trying to figure out why relayhost wasn't getting defined in main.cf. This appears to be why.

**Comment by sandynomad** [comment]
Bump for this issue. It's defined in the documentation yet the implementation doesn't match the documentation.

**Comment by rbrewster-uwm** [comment]
Still an issue in 2025. 

> True, but we don't have `postfix::mta` in our catalog, which seemingly isn't mandatory. Including that - or `postfix::satellite`, which does the same - just to have `relayhost` included in `main.cf` cannot be the advised solution, right?
